src/integrations/front-end/redirects.php

```diff
@@ -240,4 +241,27 @@ private function do_date_redirect( $query_vars, $base_url ) {
 
 		$this->redirect->do_safe_redirect( $url, 301 );
 	}
+
+	/**
+	 * Strips `cat=-1` from the URL and redirects to the resulting URL.
+	 */
+	public function category_redirect() {
+		/**
+		 * Allows the developer to keep cat=-1 GET parameters
+		 *
+		 * @since 19.9
+		 */
+		$should_remove_parameter = \apply_filters( 'wpseo_remove_cat_parameter', true );
```

Comment:
```suggestion
		 * Allows the developer to keep cat=-1 GET parameters
		 *
		 * @since 19.9
		 *
		 * @param bool $remove_cat_parameter Whether to remove the `cat=-1` GET parameter. Default true,
		 */
		$should_remove_parameter = \apply_filters( 'wpseo_remove_cat_parameter', true );
```